### ISSUE TITLE
How to combine multiple tower middleware?

### ISSUE THREAD

**Issue by Oodachi** [poem-web/poem#35]
Looks like this,  but the compilation fails.

```rust
route()
    .at("/hello/:name", get(hello))
    .with(AddData::new(db))
    .with(RateLimitLayer::new(5, Duration::from_secs(30)).compat())
    .with(TimeoutLayer::new(Duration::from_secs(30)).compat())
```

**Comment by sunli829** [comment]
This is caused by the incompatibility of the error type, and I have fixed it.

**Comment by sunli829** [comment]
Fixed in `v0.7.4`

**Comment by Oodachi** [comment]
> This is caused by the incompatibility of the error type, and I have fixed it.

Nice!
